mlflow/tracing/utils/__init__.py

```diff
@@ -342,7 +342,13 @@ def f():
             f"Invalid tools type {type(tools)}. Expected a list of ChatTool.",
             error_code=BAD_REQUEST,
         )
+
+    sanitized_tools = []
     for tool in tools:
-        ChatTool.validate_compat(tool)
+        if isinstance(tool, dict):
+            ChatTool.validate_compat(tool)
+            sanitized_tools.append(tool)
+        elif isinstance(tool, ChatTool):
+            sanitized_tools.append(tool.model_dump(exclude_unset=True))
```

Comment:
model_dump only exists in pydantic v2, we need to support pydantic v1 as well 😢

Reply:
Ahhh truee